### ISSUE TITLE
SQLSRV - varchar output parameters are declared as varchar(max) 

### ISSUE THREAD

**Issue by george-slastnoy** [microsoft/msphpsql#1025]
Just started to migrate my projects from PHP 5.6/Freetds driver to PHP 7.3.8/SQLSRV laters driver (5.6.1), Odbc 17. The SQL Server version is 2008 R2 SP2.

Noticed that the driver always declares varchar(n) output parameters as varchar(max), althout fixed char(n) ouput parameters are declared correctly.

**Comment by yitam** [comment]
Please provide a repro script. Just so you know, as indicated in our [documentation for retrieving output parameters](https://docs.microsoft.com/sql/connect/php/how-to-retrieve-output-parameters-using-the-sqlsrv-driver?view=sql-server-2017), 

> Data truncation can occur when stream types such as SQLSRV_SQLTYPE_VARCHAR('max') are used as output parameters. Stream types are not supported as output parameters. For non-stream types, data truncation can occur if the length of the output parameter is not specified or if the specified length is not sufficiently large for the output parameter.

**Comment by george-slastnoy** [comment]
No, I mean the following: if I call a procedure with a varchar(n) output parameter, in SQL Server Profiler this parameter is declared as varchar(max).

My function in php:
function get_city($city_id)
	{
	global $linkid;
	$city_id=intval($city_id);
	$c='';
	if ($stmt=sqlsrv_query($linkid,'{CALL [dbo].[get_city] (?,?)}',array(array($city_id),
		array(&$c,SQLSRV_PARAM_OUT,SQLSRV_PHPTYPE_STRING(SQLSRV_ENC),SQLSRV_SQLTYPE_VARCHAR(50))))) sqlsrv_free_stmt($stmt);
	return $c;
	}
The SQL Server Profiler output:
declare @p2 varchar(max)
set @p2='Иркутск'
exec [dbo].[get_city] 2,@p2 output
select @p2

**Comment by george-slastnoy** [comment]
Interesting. If it is by design, then why specify the length of parameters at all (SQLSRV_SQLTYPE_VARCHAR(length))? And in SQL Server, does it make no difference with what length to declare a variable?

**Comment by yitam** [comment]
Closing this issue @george-slastnoy. Please feel free to reopen if necessary.
